### ISSUE TITLE
Update tensorflow-hub requirement to allow 0.12.0?

### ISSUE THREAD

**Issue by suryaavala** [tensorflow/tfx#3812]
If the feature is related to a specific library below, please raise an issue in
the respective repo directly:

[TensorFlow Data Validation Repo](https://github.com/tensorflow/data-validation/issues)

[TensorFlow Model Analysis Repo](https://github.com/tensorflow/model-analysis/issues)

[TensorFlow Transform Repo](https://github.com/tensorflow/transform/issues)

[TensorFlow Serving Repo](https://github.com/tensorflow/serving/issues)

**System information**

- TFX Version (you are using):  1.0.0-rc0
- Environment in which you plan to use the feature (e.g., Local
  (Linux/MacOS/Windows), Interactive Notebook, Google Cloud, etc..): MacOS, AWS
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
tfx (1.0.0-rc0) currently depends on tensorflow-hub (>=0.9.0,<0.10)

I was wondering if we could update tensorflow-hub dependancy for tfx to allow tf-hub 0.12.0, so something like (>=0.9.0,<=0.12.0)?

I am not sure if that would break anything in tfx, but I am happy to investigate and contribute to this change

**Will this change the current API? How?**
No

**Who will benefit with this feature?**
tensorflow-hub has added some new features in 0.10.0 and beyond (specifically the one I'm interested in "`compute_output_shape` in `hub.KerasLayer`" which they added in 0.12.0). It would be cool to be able to take advantage of those while still being able to use tfx

**Do you have a workaround or are completely blocked by this?** :
Blocked

**Name of your Organization (Optional)**

**Any Other info.**

**Comment by chongkong** [comment]
Thanks @suryaavala. We missed to keep the dependency constraint up to date.

**Comment by suryaavala** [comment]
No worries @chongkong! 

I've already opened a PR ([#3813](https://github.com/tensorflow/tfx/pull/3813)) with the issue to update the dependency. Looks like you might have missed it and created another one. Should I close mine in favour of yours?

**Comment by chongkong** [comment]
Oh sorry I missed your PR. Would you just update the release note as well as I did? Will use your's.
